### DISCUSSION TITLE
How to set global field on logger

### DISCUSSION THREAD

**Discussion by OrkhanAlikhanov** [goravel/goravel#281] [Q&A]
Laravel has a feature where it allows you to set error field for all the subsequent calls. This is useful for adding fields like `request-id` https://laravel.com/docs/10.x/logging#contextual-information

Obviously such information needs to be stored in `ctx`. But I could not find an easy way to do this in goravel.

I have a workaround though. I set global `"request-id"` in my middleware:
```go
func HttpLogger() http.Middleware {
	return func(ctx http.Context) {
		ctx.WithValue("request-id", uuid.New().String())
		ctx.Request().Next()
	}
}
```

Use that `"request-id"` in my `LogEntry` utility:
```go
func LogEntry(ctx context.Context) log.Writer {
	return facades.Log().WithContext(ctx).With(map[string]any{"request-id": ctx.Value("request-id")})
}
```

And in my code l use my `logging.LogEntry(ctx)` each time I need to log something which will log `request-id`.

**Comment by hwbrzzl** [comment]
Make sense, we may add the Context value to Log when calling `WithContext`.

**Reply by hwbrzzl** [reply]
Added a new issue: https://github.com/goravel/goravel/issues/282
